add prevpow2 and nextpow2 benchmarks

rfourquet opened the pull request:
WIP as I'm not familiar at all with the framework. @tkelman [asked me](https://github.com/JuliaLang/julia/pull/21231#issuecomment-290355981) if benchmarks exist for those function, here they are. I'm not sure about pretty much everything:

1. the `tol` parameter, which I copied verbatim from other benchmarks;
2. the name "intfuncs" of the group (after the base/intfuncs.jl file where the functions are defined), and the choice of the "scalar" group;
3. the name of `tstr` containing a sign (e.g. "BigInt +") (the perfs depend on the sign of the argument);
4. whether to test multiple values, to get a sort of average, instead of only one value.

jrevels commented on src/scalar/ScalarBenchmarks.jl:
```diff
@@ -161,4 +161,21 @@ for b in values(g)
+        g["prevpow2", tstr] = @benchmarkable map(prevpow2, $(sgn(x))) time_tolerance=tol
+        g["nextpow2", tstr] = @benchmarkable map(nextpow2, $(sgn(x))) time_tolerance=tol
+    end
+end
```
I would change this to the following:

```julia
for T in INTS
    x = T[0, 1, 2, 3, 4, 10, 100, 1000, 10000, 100000]
    y = similar(x)
    tol = in(T, BIGNUMS) ? 0.40 : 0.25
    tstr = string(T)
    for sgn in (+, -)
        sgnstr = string(sgn)
        g["prevpow2", tstr, sgnstr] = @benchmarkable map!(prevpow2, $y, $(sgn(x))) time_tolerance=tol
        g["nextpow2", tstr, sgnstr] = @benchmarkable map!(nextpow2, $y, $(sgn(x))) time_tolerance=tol
    end
end
```

The two important changes here are:

- splitting up the type and sign in the benchmark ID
- Using in-place `map!` instead of allocating a temporary for each benchmark execution. This should cut down on noise and makes the benchmark more targeted (i.e. it avoids benchmarking allocation overhead).

rfourquet commented:
Thank you for the review! I updated according to your request and added some new values to test. Also I included the function to test as a for-loop variable, which is not in the style of the file, so I can revert that bit.

jrevels commented:
For future reference, it's usually cleaner to manually unroll such loops when the number of combined items are so small and are known at program time. It doesn't matter enough to hold up this PR though. Thanks again for the contribution!

rfourquet commented:
Ok and thank you!
